### ISSUE TITLE
`flux trace` fails executing template

### ISSUE THREAD

**Issue by makkes** [fluxcd/flux2#1621]
### Describe the bug

```sh
$ flux trace my-cm -n some-ns --kind configmap --api-version=v1
✗ template: tmpl:28:21: executing "tmpl" at <.GitRepository.Spec.Reference.Tag>: nil pointer evaluating *v1beta1.GitRepositoryRef.Tag
```

### To Reproduce

Steps to reproduce the behaviour:

1. `flux install`
1. ```sh
   cat <<EOF | kubectl apply -f-
   apiVersion: source.toolkit.fluxcd.io/v1beta1
   kind: GitRepository
   metadata:
     name: podinfo
     namespace: flux-system
   spec:
     gitImplementation: go-git
     interval: 1m0s
     timeout: 20s
     url: https://github.com/stefanprodan/podinfo
   EOF
   ```
1. ```sh
   cat <<EOF | kubectl apply -f-
   apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
   kind: Kustomization
   metadata:
     name: podinfo
     namespace: flux-system
   spec:
     interval: 5m0s
     path: ./kustomize
     prune: true
     sourceRef:
       kind: GitRepository
       name: podinfo
     validation: client
   EOF
   ```

1. ```sh
   flux trace -n default podinfo --kind deployment --api-version=apps/v1
   ✗ template: tmpl:28:21: executing "tmpl" at <.GitRepository.Spec.Reference.Tag>: nil pointer evaluating *v1beta1.GitRepositoryRef.Tag
   ```

### Expected behavior

I would expect the resource trace to be printed.

### Additional context

- Kubernetes version: 1.21.1
- Git provider: GitHub
- Container registry provider: n/a

Below please provide the output of the following commands:

```cli
flux --version
flux version 0.16.1
flux check
► checking prerequisites
✔ kubectl 1.21.2 >=1.18.0-0
✔ Kubernetes 1.21.1 >=1.16.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.11.1
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.13.2
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.15.0
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.15.3
✔ all checks passed
```

**Comment by kingdonb** [comment]
It looks like your report is missing a step, (I inferred this part, as Step 2.5) from the getting started guide:


```
$ cat <<EOF | kubectl apply -f-
> apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
kind: Kustomization
metadata:
  name: podinfo
  namespace: flux-system
spec:
  interval: 5m0s
  path: ./kustomize
  prune: true
  sourceRef:
    kind: GitRepository
    name: podinfo
  validation: client
EOF
```

I was indeed able to reproduce the issue this way, with Flux v0.16.1. I have not looked at the `flux trace` command before but this seems to be a pretty undifferentiated use, as an example from the `flux trace` guide it matches pretty exactly.

```
$ flux trace -n default podinfo --kind deployment --api-version=apps/v1
✗ template: tmpl:28:21: executing "tmpl" at <.GitRepository.Spec.Reference.Tag>: nil pointer evaluating *v1beta1.GitRepositoryRef.Tag
```

I looked at the GitRepository again, and noted that `spec.ref` was indeed missing, which is allowed, and should be inferred as the default ref, (I think that's simply the default branch.) I updated the GitRepository to have an explicit:

```
spec:
  ref:
    branch: main
```

This solved it. The output of flux trace looks like this again:

```
$ flux trace -n default podinfo --kind deployment --api-version=apps/v1

Object:        deployment/podinfo
Namespace:     default
Status:        Managed by Flux
---
Kustomization: podinfo
Namespace:     flux-system
Path:          ./kustomize
Revision:      master/627d5c4bb67b77185f37e31d734b085019ff2951
Status:        Last reconciled at 2021-07-14 14:52:20 -0400 EDT
Message:       Applied revision: master/627d5c4bb67b77185f37e31d734b085019ff2951
---
GitRepository: podinfo
Namespace:     flux-system
URL:           https://github.com/stefanprodan/podinfo
Branch:        master
Revision:      master/627d5c4bb67b77185f37e31d734b085019ff2951
Status:        Last reconciled at 2021-07-14 14:54:24 -0400 EDT
Message:       Fetched revision: master/627d5c4bb67b77185f37e31d734b085019ff2951
```

Looks like a bug.

**Comment by makkes** [comment]
Thanks @kingdonb! Yes, I forgot the Kustomization creation step. 🤦🏻‍♂️ I updated the issue description accordingly. Yes, I agree that the optional `Reference` isn't taken into account.

**Comment by yebyen** [comment]
It looks to be resolved in #1626 👍
